File: Frogs and Jumps[26-March-2023]/Solution.java
//		>> JAVA CODE << 

class Solution {
    public int unvisitedLeaves(int N, int leaves, int frogs[]) {
        
        boolean[] visited=new boolean[leaves+1];
        
        for(int frog:frogs)
        {
            if(frog<=leaves && visited[frog]) continue;
            
            for(int i=frog;i<=leaves;i+=frog)
            {
               visited[i]=true;
            }
        }
        
        int cnt=0;
        
        for(int i=1;i<leaves+1;i++)
        {
            if(visited[i]==false) cnt++;
        }
        
        return cnt;
    }
}

File: Frogs and Jumps[26-March-2023]/solution.cpp
//			>> C++ CODE <<

class Solution {
  public:
    int unvisitedLeaves(int N, int leaves, int frogs[]) {
        
       vector<bool> visited(leaves + 1, false);
    int cnt = 0;
    for (int i = 0; i < N; i++) {
        int frog = frogs[i];
        if (frog <= leaves && visited[frog]) {
            continue;
        }
        for (int j = frog; j <= leaves; j += frog) {
            visited[j] = true;
        }
    }
    for (int i = 1; i <= leaves; i++) {
        if (!visited[i]) {
            cnt++;
        }
    }
    return cnt;
    }
};